src/commands/build/build.js

```diff
@@ -7,11 +7,7 @@ const { error, exit, generateNetlifyGraphJWT, getEnvelopeEnv, getToken, normaliz
 /**
  * @param {import('../../lib/build').BuildConfig} options
  */
-const checkOptions = ({ cachedConfig: { siteInfo = {} }, token }) => {
-  if (!siteInfo.id) {
-    error('Could not find the site ID. Please run netlify link.')
-  }
-
+const checkOptions = ({ token }) => {
   if (!token) {
```

Comment:
Does this mean it won't work unless logged-in?

Reply:
This is only run `if (!options.offline)` - we are still making API requests with a user's account, and filling in config from that. This will grab any plugins they have installed that aren't just the ones from framework-info. 

I think changing this is out of scope for this PR, but we could potentially re-think how we're using offline/online mode, and whether or not we error out, or just warn.

Reply:
Makes sense